compiler-core/src/type_/tests/errors.rs

```diff
@@ -105,7 +105,7 @@ macro_rules! assert_error {
 }
 
 macro_rules! assert_with_module_error {
-    ($(($name:expr, $module_src:literal)),+, $src:expr $(,)?) => {
+    (($name:expr, $module_src:literal), $src:expr $(,)?) => {
```

Comment:
I tried fixing the variadic macro but I couldn't. I unrolled it and implemented the new tests.